### ISSUE TITLE
Unknown transaction resets staking period

### ISSUE THREAD

**Issue by workvint** [EthereumCommonwealth/Cold-staking#15]
**About 29 days ago I did** [deposit](https://explorer2.callisto.network/tx/0x9025a6f28d7a161e25406b605518fbc5b71cb569ad2a790de4163da915905e06) to staking contract  **17130** CLO (after that, there are about  122000 coins in the contract).
Staking period must be ended 5 january 2019. During process was error about gas required exceeds.

31 december 2018 (5 days before staking period must end) I send to my wallet 13980 CLO from HitBtc. 
After that, there are about  19000 coins in the wallet. 

After a couple of minutes, a **transaction appeared that sent** exactly **17130** CLO to the staking contract. 
It looks as if [this transaction](https://explorer2.callisto.network/tx/0x947d9f214c11ed192a423bfde1e327f77f9dce9e5e33c5ed1dbbcb88f5303370) was created at the beginning of the period (that is, 22 days ago) and was expecting the necessary amount on the wallet to be executed.

So, [this transaction](https://explorer2.callisto.network/tx/0x947d9f214c11ed192a423bfde1e327f77f9dce9e5e33c5ed1dbbcb88f5303370) *which I did not send* has reset staking period and reward.

---

I would not want this to happen again. Not only because I lost about 15,000 coins as a reward for the second round, but even so such "dormant transactions" are not credible.

**Comment by workvint** [comment]
@RideSolo I used https://wallet.callisto.network/ for staking

Perhaps I did not explain well. I have no complaints about the smart contract.

I do not understand where the second transaction came from with the same amount of coins that was executed when there were enough funds on the wallet. It is very strange.

I waited for the period to end (4 days left) to add coins and start a new period. And then after refill of the wallet occurs this transaction.

Perhaps, two transactions could be created in the [wallet](https://wallet.callisto.network/) (nonce 8 and 9), one was executed immediately 29 days ago, and the second was waiting for funds. Is it possible?

**Comment by workvint** [comment]
@RideSolo 
It is possible to find out the date of the creation of the transaction (pending), 
**not the date when it was mining and recorded in the block**, but the date when the transaction was sent to the network (created)?

**Comment by RideSolo** [comment]
@workvint I do understand you but it has to be clear for everyone, if a transaction is made and propagated to the network and if there was not enough of fund to cover it, the transaction will fail.

It is not possible to get the date of creation.

did you close the wallet ? after sending the first transaction. I mean ur browser in general

**Comment by workvint** [comment]
@RideSolo firefox was in incognito mode, PC was shutdown. It was 29 days ago, so yeah.

That's why I didn’t want to open this issue, since from the outside everything looks as if I had sent a second transaction.

**Comment by RideSolo** [comment]
@workvint yeah it is a really weird issue, and I do trust you I will check with the team about wallet.callisto.network but I'm more than sure that the wallet runs in ur browser only and nothing is saved on the server side.

**Comment by workvint** [comment]
@RideSolo I managed to reproduce this bug on another wallet:

* Firefox 64.0, private mode; https://wallet.callisto.network/ gas price 2 gwei.
* There should be enough funds on wallet ABC for just one transaction per staking contract (e.g. 3.95 CLO)
* We perform stacking by pressing the button **Yes, I am sure! Make transaction.** very quickly several times (send 2 CLO).
* Only the first transaction is performed. And the contract gets 2 coins.
* Send to the wallet ABC 1 CLO from other wallet.
* From about 1 minutes after 1 CLO is arrived to wallet ABC a transaction appeared that sent exactly 2 CLO to the staking contract.

**Comment by RideSolo** [comment]
@workvint it can be the RPC node that check if u have sufficient fund for transfer, maybe ur transaction stayed pending on that node and did not propagate, I have asked the team already. we will check everything and let you know.

**Comment by workvint** [comment]
> I have asked the team already. we will check everything and let you know.

Any news from the dev team?
